docker/volume/volume.go

```diff
@@ -0,0 +1,154 @@
+package volume
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/engine-api/client"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/libcompose/config"
+
+	composeclient "github.com/docker/libcompose/docker/client"
+	"github.com/docker/libcompose/project"
+)
+
+// Volume holds attributes and method for a volume definition in compose
+type Volume struct {
+	client        client.VolumeAPIClient
+	projectName   string
+	name          string
+	driver        string
+	driverOptions map[string]string
+	external      bool
+	// TODO (shouze) missing labels
+}
+
+func (v *Volume) fullName() string {
+	name := v.projectName + "_" + v.name
```

Comment:
Yeah I did the same for networks. I'll think about it and probably do some follow-up later about that. It's tricky because the 2 objects manipulated here and in `handleVolumeConfig` are different (one is a service volume, the other one is the declaration of the volume).